sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/subquery.scala

```diff
@@ -77,3 +77,24 @@ case class ScalarSubquery(
 
   override def toString: String = s"subquery#${exprId.id}"
 }
+
+/**
+ * Base interface for correlated subquery expressions.
+ */
+abstract class CorrelatedSubqueryExpression extends SubqueryExpression{
```

Comment:
space after SubqueryExpression

For the classdoc, we should document that this gets replaced in the analyzer